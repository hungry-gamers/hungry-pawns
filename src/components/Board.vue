<script setup lang="ts"></script>

<template>
  <div id="board">
    <div class="cell"></div>
  </div>
</template>

<style scoped>
#board {
  display: grid;
  grid-template-columns: repeat(3, 100px);
  grid-template-rows: repeat(3, 100px);
  gap: 5px;
}

.cell {
  width: 100px;
  height: 100px;
  background-color: white;
  border: 2px solid black;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 2rem;
  cursor: pointer;
}
</style>
