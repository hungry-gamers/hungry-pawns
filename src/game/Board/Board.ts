export class Board {
  static size = '3x3'

  constructor() {}

  calculateCells() {
    const [rows, columns] = Board.size.split('x')

    return +rows * +columns
  }
}
