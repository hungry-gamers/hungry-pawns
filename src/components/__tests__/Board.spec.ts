import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import Board from '../Board.vue'

describe('Board', () => {
  const wrapper = mount(Board)

  it('renders properly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
