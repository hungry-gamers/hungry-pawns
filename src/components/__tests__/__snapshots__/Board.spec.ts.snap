// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Board > renders properly 1`] = `
"<div data-v-93a62ddb="" id="board">
  <div data-v-93a62ddb="" class="cell"></div>
</div>"
`;
